radix-engine/src/engine/process.rs

```diff
@@ -67,12 +79,19 @@ pub struct Process<'r, 'l, L: SubstateStore> {
     moving_buckets: HashMap<Bid, Bucket>,
     /// The bucket refs that will be moved to another process SHORTLY.
     moving_bucket_refs: HashMap<Rid, BucketRef>,
-    /// Lazy maps which haven't been assigned to a component or lazy map yet.
-    unclaimed_lazy_maps: HashMap<Mid, LazyMap>,
+
     /// Vaults which haven't been assigned to a component or lazy map yet.
     unclaimed_vaults: HashMap<Vid, Vault>,
+    /// Vaults which have been assigned to a lazy map but not a component
+    claimed_vaults: HashMap<Vid, (Vault, Mid)>,
+    /// Lazy maps which haven't been assigned to a component or lazy map yet.
+    unclaimed_lazy_maps: HashMap<Mid, (LazyMap, HashSet<Mid>, HashSet<Vid>)>,
```

Comment:
Suggest to have a structure with named fields for the value tuple. To some extend, we should probably have a forest representation.